controllers/agents/svc/serviceclass_controller.go

```diff
@@ -102,7 +104,21 @@ func (r *ServiceClassReconciler) Reconcile(ctx context.Context, req ctrl.Request
 		return ctrl.Result{}, err
 	}
 
-	if err = r.setOwnerReference(ctx, &serviceClass, req.Namespace); err != nil {
+	// get the controller's deployment
+	controller := appsv1.Deployment{}
+	controllerRef := types.NamespacedName{Namespace: serviceClass.Namespace, Name: constants.ServiceAgentDeploymentName}
+	if err = r.Get(ctx, controllerRef, &controller); err != nil {
+		reconcileLog.Error(err, "Failed to retrieve controller reference")
+        if apierrors.IsNotFound(err) {
+            // FIXME(sadlerap): the deployment's been deleted, and the pod
+            // we're running in is likely going to be deleted soon as well.  Do
+            // we have a cleaner way of triggering our own shutdown?
+            os.Exit(1)
+        }
```

Comment:
indentation fix and I'd propose to add a log line here

```suggestion
                if apierrors.IsNotFound(err) {
                    // FIXME(sadlerap): the deployment's been deleted, and the pod
                    // we're running in is likely going to be deleted soon as well.  Do
                    // we have a cleaner way of triggering our own shutdown?
                    reconcileLog.Error(err, "can not find agent's deployment", "agent", constants.ServiceAgentDeploymentName)
                    os.Exit(1)
                }
```

Reply:
PTAL at #156